File: 프로그래머스/카카오 기출문제/2023 KAKAO BLIND RECRUITMENT/택배 배달과 수거하기.js
function solution(cap, n, deliveries, pickups) {
	const D = deliveries.map((num, index) => Array(num).fill(index + 1)).flat();
	const P = pickups.map((num, index) => Array(num).fill(index + 1)).flat();

	let dist = 0;
	while (D.length || P.length) {
		const farthestD = D.length ? D[D.length - 1] : 0;
		const farthestP = P.length ? P[P.length - 1] : 0;
		const farthest = Math.max(farthestD, farthestP);

		dist += farthest * 2;

		for (let i = 0; i < cap; i++) {
			if (D.length) D.pop();
			if (P.length) P.pop();
		}
	}

	return dist;
}

File: 프로그래머스/카카오 기출문제/2023 KAKAO BLIND RECRUITMENT/택배 배달과 수거하기.py
def solution(cap, n, deliveries, pickups):
    D = [index+1 for index, num in enumerate(deliveries) for _ in range(num)]
    P = [index+1 for index, num in enumerate(pickups) for _ in range(num)]
        
    dist = 0
    while D or P:
        farthest = max((D[-1] if D else 0), (P[-1] if P else 0))
        dist += farthest * 2
        for _ in range(cap):
            if D: D.pop()
            if P: P.pop()
        
    return dist